src/service/variable-source.js

```diff
@@ -44,15 +80,40 @@ const LOAD_EVENT_END = 'loadEventEnd';
  * @return {!Promise<ResolverReturnDef>}
  */
 export function getTimingDataAsync(win, startEvent, endEvent) {
-  let readyPromise = loadPromise(win);
-  if (startEvent === LOAD_EVENT_END || endEvent === LOAD_EVENT_END) {
+  // Fallback to load event if we don't know what to wait for
+  const startWaitForEvent =
+    NAV_TIMING_WAITFOR_EVENTS[startEvent] || WAITFOR_EVENTS.LOAD;
+  const endWaitForEvent = endEvent
+    ? NAV_TIMING_WAITFOR_EVENTS[endEvent] || WAITFOR_EVENTS.LOAD
+    : null;
+
+  let waitForEvent;
+  if (!endWaitForEvent) {
+    waitForEvent = startWaitForEvent;
+  } else {
+    waitForEvent = Math.max(startWaitForEvent, endWaitForEvent);
+  }
+
+  // set wait for onload to be default
+  let readyPromise;
+  if (waitForEvent === WAITFOR_EVENTS.VIEWER_FIRST_VISIBLE) {
+    readyPromise = Promise.resolve();
+  } else if (waitForEvent === WAITFOR_EVENTS.DOCUMENT_COMPLETE) {
+    readyPromise = whenDocumentComplete(win.document);
+  } else if (waitForEvent === WAITFOR_EVENTS.LOAD) {
+    readyPromise = loadPromise(win);
+  } else if (waitForEvent === WAITFOR_EVENTS.LOAD_END) {
     // performance.timing.loadEventEnd returns 0 before the load event handler
     // has terminated, that's when the load event is completed.
     // To wait for the event handler to terminate, wait 1ms and defer to the
     // event loop.
     const timer = Services.timerFor(win);
-    readyPromise = readyPromise.then(() => timer.promise(1));
+    readyPromise = loadPromise(win).then(() => timer.promise(1));
+  } else {
+    dev().error('NAV-TIMING', 'waitForEvent not found %s', waitForEvent);
```

Comment:
Is this user controlled at all? If so, can we make this a `dev().assert`?

Reply:
Good point. switch to devAssert.